### PR TITLE
Vignettes roundup

### LINKED ISSUE #65: vignettes roundup
how/why to prebuild them (html strategies, maybe for PDF too)

e.g. https://community.rstudio.com/t/precompiling-vignette-with-devtools/1583

https://ropensci.org/technotes/2019/12/08/precompute-vignettes/

http://www.markvanderloo.eu/yaRb/2019/01/11/add-a-static-pdf-vignette-to-an-r-package/ and https://www.mail-archive.com/r-package-devel@r-project.org/msg04642.html

Not sure if that's enough for a post. Maybe it's possible to detect strategies in existing CRAN packages.

### PR DESCRIPTION
**Pull request by maelle**
fix #65

### CODE REVIEW

File: content/post/2020-06-03-vignettes/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+The building of package vignettes can either use the default Sweave vignette engine, or [a vignette engine provided by a CRAN package](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Non_002dSweave-vignettes) like [`knitr` by Yihui Xie](https://bookdown.org/yihui/rmarkdown-cookbook/package-vignette.html).
+[`knitr::rmarkdown` vignette engine](https://community.rstudio.com/t/question-about-usethis-vignette-template/32048) is the one recommended in the R packages book, and `usethis`.
+It allows writing vignettes in R Markdown.
+
+[See the source of `rhub` main vignette](https://github.com/r-hub/rhub/blob/master/vignettes/rhub.Rmd). 
+It has YAML metadata at the top, some non-executed code chunks, some executed code chunks.
+To allow for that vignette to be built, a [field in `DESCRIPTION`](https://github.com/r-hub/rhub/blob/6ae6f35e958f3beab1e2c8e6f704affa23c8ce29/DESCRIPTION#L47) mentions the vignette engine:
+
+```yaml
+VignetteBuilder: knitr
```
[the remark this answers is not in the document]
Comment (maelle) [reply content/post/2020-06-03-vignettes/index.Rmd:100]: Why?

[the remark this answers is not in the document]
Comment (gaborcsardi) [reply content/post/2020-06-03-vignettes/index.Rmd:100]: https://github.com/wch/r-source/blob/0b3747da1bc1338d287b2406c221040a7b1637dc/src/library/tools/R/check.R#L210

Only if you are using rmarkdown, of course.

[the remark this answers is not in the document]
Comment (maelle) [reply content/post/2020-06-03-vignettes/index.Rmd:100]: oh! and listing rmarkdown in Suggests https://github.com/r-hub/rhub/blob/6ae6f35e958f3beab1e2c8e6f704affa23c8ce29/DESCRIPTION#L45 isn't enough/the right way then?

[the remark this answers is not in the document]
Comment (gaborcsardi) [reply content/post/2020-06-03-vignettes/index.Rmd:100]: > Note that if, for example, a vignette has engine ‘knitr::rmarkdown’, then knitr provides the engine but both knitr and rmarkdown are needed for using it, so both these packages need to be in the ‘VignetteBuilder’ field and at least suggested (as rmarkdown is only suggested by knitr, and hence not available automatically along with it). Many packages using knitr also need the package formatR which it suggests and so the user package needs to do so too and include this in ‘VignetteBuilder’.

https://cran.r-project.org/doc/manuals/R-exts.html#The-DESCRIPTION-file

[the remark this answers is not in the document]
Comment (maelle) [reply content/post/2020-06-03-vignettes/index.Rmd:100]: ooooh I totally missed that! To my defence it wasn't in the vignettes section but I suppose I should read the whole thing. Thank you!!

File: content/post/2020-06-03-vignettes/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+### How to include my pre-print / cheatsheet as a PDF vignette?
+
+Sometimes you might want to include a PDF as a vignette, without wanting to deal with missing LaTeX dependencies; or because the PDF is not knit from R (a cheatsheet); or the computations are too long. 
+In that case there are two alternatives:
+
+* Following the process described in [a blog post by 
+Mark van der Loo, entitled _"Add a static pdf vignette to an R package"_](http://www.markvanderloo.eu/yaRb/2019/01/11/add-a-static-pdf-vignette-to-an-r-package/);
+
+* Using the [`R.rsp` package](https://cran.r-project.org/web/packages/R.rsp/index.html) by Henrik Bengtsson.
```
[the remark this answers is not in the document]
Comment (maelle) [reply content/post/2020-06-03-vignettes/index.Rmd:182]: right but I'm guessing many people using such workarounds already have LaTeX installed?

**Pull request merged**